fix(RELEASE-1908): Replace deprecated conforma URL

elenagerman opened the pull request:
## Describe your changes
Replaced https://github.com/enterprise-contract/ec-cli with the https://github.com/conforma/cli since it was "moved permanently" to the new location.

Note: 
This particular change has an indirect connection to the Conforma change. When the Conforma CLI URL was updated across multiple files, the CI validation script automatically ran on all changed files to check security compliance. It caught the missing security context, which was then fixed in the same commit to maintain compliance.

## Relevant Jira
[RELEASE-1908](https://issues.redhat.com/browse/RELEASE-1908)

Signed-off-by: elenagerman [elgerman@redhat.com](mailto:elgerman@redhat.com)

## Checklist before requesting a review
- [x] I have marked as draft or added `do not merge` label if there's a dependency PR
  - If you want reviews on your draft PR, you can add reviewers or add the `release-service-maintainers` handle if you are unsure who to tag
- [x] My commit message includes `Signed-off-by: My name <email>`
- [x] I read CONTRIBUTING.MD and [commit formatting](CONTRIBUTING.md#commit-message-formatting-and-standards)
- [ ] I have run the README.md generator script in `.github/scripts/readme_generator.sh` and verified the results using `.github/scripts/check_readme.sh`

elenagerman committed:
fix(RELEASE-1908): replace deprecated conforma URL

Replaced https://github.com/enterprise-contract/ec-cli with the
https://github.com/conforma/cli since it was "moved permanently"
to the new location.

Signed-off-by: Elena German <elgerman@redhat.com>

elenagerman commented:
/retest

elenagerman commented:
@FilipNikolovski Sorry, a little bit more modification was required related to the quay.io/enterprise-contract/ec-cli swap with the quay.io/conforma/cli image; a different architectural approach to containerized CLI tools

FilipNikolovski reviewed (commented):
I see a lot more changes in the tests, why are those required, shouldn't this be a drop-in replacement?

FilipNikolovski reviewed (dismissed):
lgtm

mmalina commented on tasks/managed/check-data-keys/check-data-keys.yaml:
```diff
@@ -78,6 +78,8 @@ spec:
         value: "$(params.orasOptions)"
       - name: "DEBUG"
         value: "$(params.trustedArtifactsDebug)"
+    securityContext:
+      runAsUser: 1001
```
Why is this needed? How is it related to the conforma change?

elenagerman replied:
It is NOT related to the Conforma change. It's a separate security requirement for tasks that use Trusted Artifacts:
           - Security Requirement: Tasks using Trusted Artifacts (those with ociStorage parameter) must run as non-root user (1001) instead of root (0)
           - Workaround for Cluster Issue: As documented in the code comments: 
    # This is a workaround for a problem observed on a particular cluster where the
    # use-trusted-artifacts step runs with root user, causing a Docker credential file
    # will not be readable in later steps.
           - There's even a dedicated script (.github/scripts/tkn_check_task_runasuser.sh) that automatically validates this requirement ->

So, unfortunately, it has to be included in this commit since our change applies to one of the relevant files, and it triggers assigned tests and fails otherwise: 
https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/managed/publish-to-mrrc/publish-to-mrrc.yaml

mmalina replied:
> So, unfortunately, it has to be included in this commit since our change applies to one of the relevant files, and it triggers assigned tests and fails otherwise

I'm fine to include this, but can you mention this in the commit message and PR description?

elenagerman replied:
I have no idea why it was not added before, actually.
It uses Trusted Artifacts and non-release-service-utils images => it MUST have runAsUser: 1001 in stepTemplate.securityContext. Correct me if I'm wrong.

elenagerman replied:
@mmalina Another option is to create a separate MR with only the runAsUser fixes, and we'll merge it first. 
Please let me know WDYT.
Thank you!

elenagerman replied:
But the check-data-keys.yaml was the only place where it was missing. I didn't find any additional problematic files.

mmalina replied:
> It uses Trusted Artifacts and non-[release-service-utils](https://issues.redhat.com/browse/RELEASE-service-utils) images => it MUST have runAsUser: 1001 in stepTemplate.securityContext. Correct me if I'm wrong.

I am not disputing that. I said I was fine with it. But I am asking you to include it in the PR description and commit message. The commit message could use more details anyway.

> Another option is to create a separate MR with only the runAsUser fixes

Fine by me. Up to you. Again, I am fine with including it here.

elenagerman committed:
fix: add securityContext to check-data-keys.yaml

"securityContext: runAsUser: 1001" was added to
the check-data-keys.yaml.
Tasks using Trusted Artifacts (with ociStorage parameter)
must run as non-root user (1001) instead of root (0) for
security compliance.
This change addresses a specific issue where steps running
as the root user cause Docker credential files to become
unreadable in subsequent steps on particular clusters.
It's a security compliance requirement to prevent
permission issues and ensure proper file access in
Trusted Artifacts workflows

Signed-off-by: Elena German <elgerman@redhat.com>

elenagerman commented:
> > sorry, need your approval again. I moved the securityContext change to a separate commit, and it removed your original approval. Thank you!
> 
> Can you rebase and remove the merge commit please? (When rebasing via UI, I recommend using the Rebase option, not the Merge one.) Also, thanks for the note, but what I really meant is just a simple note saying that this was needed because CI was failing on it and we have that elsewhere. The commits still have no description, only title. Please add some. It's not a hard rule, but most of the time it's desirable to have more details in the commit message than just the title.

I'm not using UI to rebase. I do it from the IDE: fetch origin, rebase, push.
You don't like that the securityContext change was done in a separate commit?

`
It is generally considered good practice to have a separate commit for a change in the same MR if it is not closely related to the main change.
This concept is often summarized by the mantra: "One commit, one logical change."
`

elenagerman commented:
/retest

johnbieren commented:
> > This failed the fbc-release e2e test and was merged. As we talked about in standup yesterday, we have to be very careful and avoid this until it is fixed and shows the red X again
> 
> revert it in a meantime?

I don't see how (judging by the diff) this could have broken fbc and nothing else, so I wouldn't worry about it, but let's keep it in mind in case we see a bunch of fbc failures now that this is in
